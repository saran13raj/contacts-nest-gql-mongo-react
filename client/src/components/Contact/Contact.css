.contact {
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	align-items: center;
	/* border: solid 1px; */
	border-radius: 20px;
	padding: 10px 15px;
	margin-bottom: 10px;
	width: 200px;
	/* flex-basis: '50%'; */
	flex: 1 1 20%;
	margin: 5px;
	/* height: 50px; */
	background: #f3f3f3;
}

.contact__link {
	text-decoration: none;
	color: #000000;
	cursor: pointer;
}

.contact__pic {
}

.contact__pic--border {
	border-radius: 50%;
}

.contact__name {
	/* display: flex; */
}
