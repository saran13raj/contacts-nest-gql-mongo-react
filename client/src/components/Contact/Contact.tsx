import React from 'react';
import { Link } from 'react-router-dom';

import './Contact.css';

const Contact = ({ contact }: any) => {
	const defaultAvatar =
		'https://www.kindpng.com/picc/m/78-785827_user-profile-avatar-login-account-male-user-icon.png';

	const linkTo = {
		pathname: '/contact',
		state: { contact }
	};

	return (
		<div key={contact._id} className='contact'>
			<Link to={linkTo} className='contact__link'>
				<div className='contact__pic'>
					<img
						src={contact.picture ? contact.picture : defaultAvatar}
						alt='contact pic'
						width='60'
						height='60'
						className='contact__pic--border'
					/>
				</div>
				<div className='contact__name'>
					<p>{contact.name}</p>
				</div>
			</Link>
		</div>
	);
};

export default Contact;
