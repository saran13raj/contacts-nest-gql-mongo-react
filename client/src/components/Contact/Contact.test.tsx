import React from 'react';
import renderer from 'react-test-renderer';
import ReactDom from 'react-dom';
import { MemoryRouter } from 'react-router-dom';

import Contact from './Contact';
import { render } from '@testing-library/react';

const contact = {
	_id: '61811f2004dd629f1f4fa3fa',
	name: 'diablo',
	phone: '+911132334455',
	email: 'diablo.g.com',
	info: {
		place: 'Chennai',
		sport: 'Tennis'
	},
	address: {
		_id: '6181173fa5b3355349dafa6b',
		location: 'Chennai'
	}
};

describe('Contact component', () => {
	it('renders Contact without crashing', () => {
		// const div = document.createElement('div');
		// ReactDom.render(
		// 	<MemoryRouter>
		// 		<Contact contact={contact} />
		// 	</MemoryRouter>,
		// 	div
		// );
		// ReactDom.unmountComponentAtNode(div);

		const tree = renderer.create(
			<MemoryRouter>
				<Contact contact={contact} />
			</MemoryRouter>
		);

		// console.log(tree);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('should render contact with text diablo', () => {
		const name = 'diablo';
		const rendered = render(
			<MemoryRouter>
				<Contact contact={contact} />
			</MemoryRouter>
		);
		// console.log('check contact name',rendered);
		expect(rendered.getByText(name)).toBeTruthy();
	});
});
