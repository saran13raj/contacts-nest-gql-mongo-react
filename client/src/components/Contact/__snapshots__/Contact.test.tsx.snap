// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders Contact without crashing 1`] = `
<div
  className="contact"
>
  <a
    className="contact__link"
    href="/contact"
    onClick={[Function]}
  >
    <div
      className="contact__pic"
    >
      <img
        alt="contact pic"
        className="contact__pic--border"
        height="60"
        src="https://www.kindpng.com/picc/m/78-785827_user-profile-avatar-login-account-male-user-icon.png"
        width="60"
      />
    </div>
    <div
      className="contact__name"
    >
      <p>
        diablo
      </p>
    </div>
  </a>
</div>
`;
